lib/ecto_watch/options/watcher_options.ex

```diff
@@ -33,6 +34,11 @@ defmodule EctoWatch.Options.WatcherOptions do
 
       fields = schema_mod.__schema__(:fields)
 
+      field_map =
+        Enum.reduce(fields, %{}, fn field, acc ->
+          Map.put(acc, field, schema_mod.__schema__(:field_source, field))
+        end)
+
```

Comment:
```suggestion
      field_map = Map.new(fields, & {&1, schema_mod.__schema__(:field_source, &1)})
```

I'm actually working on a library with a custom `credo` check that looks for this sort of thing: https://github.com/cheerfulstoic/credo_unneccesary_reduce

The `Map.new` suggestion is the most recent thing I've been working on 😉

Reply:
Also realizing that we should keep the variable name to be `column_map` to be consistent with the key in the schema definition.

After thinking about it, I think `column_map` is a fine name.  In the context of defining a watcher without a schema, it's basically saying "this is what your columns will be mapped to as messages", and for schema-based watchers this is an internal detail that won't be seen.